app/namespace.go

```diff
@@ -138,179 +182,215 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
-							return errors.New("nothing to change")
-						}
-						n.Spec.AcceptedClientCa = cert
-						resourceVersion := n.ResourceVersion
-						if v := ctx.String(ResourceVersionFlagName); v != "" {
-							resourceVersion = v
-						}
-						return c.updateNamespace(
-							ctx.String(NamespaceFlagName),
-							ctx.String(RequestIDFlagName),
-							resourceVersion,
-							n.Spec,
-						)
-					},
-				}},
+				Name:  "get",
+				Usage: "get the accepted client ca certificates currently configured for the namespace",
+				Flags: []cli.Flag{
+					NamespaceFlag,
+				},
+				Action: func(ctx *cli.Context) error {
+					n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+					if err != nil {
+						return err
+					}
+					out, err := parseCertificates(n.Spec.AcceptedClientCa)
+					if err != nil {
+						return err
+					}
+					return PrintObj(out)
+				},
+			}, {
+				Name:    "add",
+				Usage:   "add a new ca accepted client ca certificate",
+				Aliases: []string{"a"},
+				Flags: []cli.Flag{
+					NamespaceFlag,
+					RequestIDFlag,
+					ResourceVersionFlag,
+					caCertificateFlag,
+					caCertificateFileFlag,
+				},
+				Action: func(ctx *cli.Context) error {
+					cert, err := readCACerts(ctx)
+					if err != nil {
+						return err
+					}
+					newCerts, err := parseCertificates(cert)
+					if err != nil {
+						return err
+					}
+					n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+					if err != nil {
+						return err
+					}
+					existingCerts, err := parseCertificates(n.Spec.AcceptedClientCa)
+					if err != nil {
+						return err
+					}
+					err = existingCerts.add(newCerts)
+					if err != nil {
+						return err
+					}
+					bundle, err := existingCerts.bundle()
+					if err != nil {
+						return err
+					}
+					if n.Spec.AcceptedClientCa == bundle {
+						return errors.New("nothing to change")
+					}
+					n.Spec.AcceptedClientCa = bundle
+					return c.updateNamespace(ctx, n)
+				},
 			}, {
-				Name:    "search-attributes",
-				Usage:   "manage search attributes used by namespace",
-				Aliases: []string{"sa"},
-				Subcommands: []*cli.Command{{
-					Name:    "add",
-					Usage:   "add a new namespace custom search attribute",
-					Aliases: []string{"a"},
-					Flags: []cli.Flag{
-						NamespaceFlag,
-						RequestIDFlag,
-						ResourceVersionFlag,
-						&cli.StringSliceFlag{
-							Name:     "search-attribute",
-							Usage:    fmt.Sprintf("flag can be used multiple times; value must be \"name=type\"; valid types are: %v", getSearchAttributeTypes()),
-							Aliases:  []string{"sa"},
-							Required: true,
-						},
+				Name:    "remove",
+				Usage:   "remove existing certificates",
+				Aliases: []string{"r"},
+				Flags: []cli.Flag{
+					NamespaceFlag,
+					RequestIDFlag,
+					ResourceVersionFlag,
+					caCertificateFlag,
+					caCertificateFileFlag,
+				},
+				Action: func(ctx *cli.Context) error {
+					cert, err := readCACerts(ctx)
```

Comment:
most of this code seems super-common between this and above. factoring?

Reply:
did some refactoring.